### PR TITLE
feat(ui): Player interaction improvements

### PR DESCRIPTION
**Pull request by prewue**
https://github.com/user-attachments/assets/39487a3f-218b-4249-b9d8-b3b2def605ef

## Description

This PR refines the redesigned player bar with a focused pass on navigation UX, artwork interaction, and slider polish for both the music player and YouTube/video player surfaces.

The main UX addition is that player metadata now behaves more like users expect from a music app: artist/channel text has a larger invisible hit area and can navigate to the related artist/channel context, while artwork can navigate to the album/video-related context where available. The interaction keeps the visual layout unchanged, adds immediate loading feedback, and preserves the player bar's hover state while navigation is in progress.

## AI Prompt (Optional)

<details>
<summary>🤖 AI Prompt Used</summary>

```
A detailed AI-assisted implementation and review process was used. No single prompt generated these changes.
```

**AI Tool:** OpenAI Codex

</details>

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 📚 Documentation update
- [x] 🎨 UI/UX improvement
- [x] ♻️ Refactoring (no functional changes)
- [ ] 🧪 Test update
- [ ] 🔧 Build/CI configuration

## Related Issues

N/A

## Changes Made

- Added player bar navigation support for artist/channel and album/artwork destinations through `PlayerBarNavigationAction` and `playerBarMusicNavigation(path:)`.
- Added `PlayerBarMetadataButton` to make artist/channel metadata clickable with a larger invisible hit area, hover feedback, accessibility button traits, and loading feedback.
- Added `PlayerBarArtworkView` and `PlayerBarArtworkGlow` to centralize artwork rendering, hover overlays, loading state, shadows/glow, and shared artwork interaction behavior.
- Added `PlayerBarSliderVisuals` to share slider colors, thumb sizing, hover/drag animation values, and hit-area metrics between horizontal and vertical player sliders.
- Updated `PlayerBar` and `YouTubePlayerBar` to use the shared artwork, metadata, navigation, and slider components.
- Updated music navigation containers so player bar actions can push artist and album destinations from the current view context.
- Refined marquee title behavior and slider interaction polish while preserving the existing player/video controls.

## Testing

- [x] Unit tests pass (`xcodebuild test -only-testing:KasetTests`)
- [x] Manual testing performed
- [x] UI tested on macOS 26+

Tested with:

- `swiftformat --lint .`
- `swiftlint lint --strict --reporter github-actions-logging`
- `swift test -q --skip KasetUITests --disable-xctest` — 1541 tests passed
- `KASET_SIGNING=adhoc Scripts/build-app.sh debug`
- `Scripts/verify-release-app.sh .build/app/Kaset.app`
- Targeted UI tests for `PlayerBarUITests` and `VideoUITests`

Notes:

- `VideoUITests` passed 11/11 locally.
- The main targeted `PlayerBarUITests` run passed 17/18 with one local XCUITest flake waiting for `sidebar.home` to become hittable; the isolated rerun of `PlayerBarUITests/testPlayerBarVisibleWithCurrentTrack` passed.
- Artist/channel and artwork navigation UX was also manually verified because asserting real detail navigation through UI mocks would be fragile around mock detail-loading timing.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have run `swiftlint --strict && swiftformat .`
- [ ] I have added tests that prove my fix/feature works
- [x] New and existing unit tests pass locally
- [x] I have updated documentation if needed
- [x] I have checked for any performance implications
- [x] My changes generate no new warnings

## Screenshots

N/A

## Additional Notes

This follow-up intentionally keeps the scope limited to player bar UI/UX, shared player components, and navigation propagation needed for artist/artwork interactions.

### CODE REVIEW

**Comment by prewue** [comment]
Follow-up update: added a playback-loading shimmer state for the player progress lane. Music now derives the loading UI from real PlayerService loading/buffering states, while YouTube exposes an observable playback loading flag that is set during real video loads/reloads and cleared on playback updates. The progress fill/thumb switch to neutral loading colors and the shimmer respects light/dark theme and Reduce Motion.

**Comment by prewue** [comment]
Implemented a local UI-level stabilization for playback scrubbing.

What changed:
- Added a small `PlayerBarSeekHold` helper that keeps the displayed seek target after the user commits a scrub.
- Wired it into the main player bar, mini player, and YouTube player bar.
- The UI now holds the clicked/dragged position until the observer catches up, the value is confirmed, the track changes, or a short timeout expires.

Why this approach:
- This keeps the change scoped to the player UI and avoids touching playback service semantics in this follow-up.
- It prevents stale observer progress from visually snapping the thumb back immediately after seek.

Future global approach:
- Move this concept into `PlayerService` / `YouTubePlayerService` as a pending seek target.
- Filter stale observer updates at the source of truth so every consumer of `progress` gets stable state, not only the player bars.

File: Sources/Kaset/Views/PlayerBarSeekHold.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        self.target = nil
+        self.issuedAt = nil
+    }
+
+    @discardableResult
+    mutating func clearIfCurrent(_ token: UUID) -> Bool {
+        guard self.target != nil, self.token == token else { return false }
+        self.clear()
+        return true
+    }
```
[the remark this answers is not in the document]
Comment (sozercan) [reply Sources/Kaset/Views/PlayerBarSeekHold.swift]: Addressed in f775010 by adding direct Swift Testing coverage for `PlayerBarSeekHold`: begin/display behavior, negative-target clamping, minimum confirmation age, confirmation tolerance, timeout clearing, and `clearIfCurrent` ID matching.

File: Sources/Kaset/Views/PlayerBarSeekHold.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import Foundation
+
+// MARK: - PlayerBarSeekHold
+
+struct PlayerBarSeekHold {
+    static let timeout: Duration = .milliseconds(2200)
+
+    private static let timeoutSeconds: TimeInterval = 2.2
+    private static let minimumConfirmationAgeSeconds: TimeInterval = 0.35
+    private static let confirmationTolerance: TimeInterval = 1.25
```
[the remark this answers is not in the document]
Comment (sozercan) [reply Sources/Kaset/Views/PlayerBarSeekHold.swift]: Addressed in f775010 by making the timeout duration and reconcile timeout derive from a single `timeoutMilliseconds` source of truth.

File: Sources/Kaset/Views/PlayerBar.swift [hunk 3]
```diff
@@ -709,6 +875,159 @@ struct PlayerBar: View {
...
+            self.resolvedArtist = artist
+        }
+
+        if self.currentAlbumPlaylist == nil,
+           let track = self.playerService.currentTrack
+        {
+            let album = await self.resolveAlbum(for: track, client: client)
+            guard !Task.isCancelled, self.currentTitleIdentity == identity else { return }
+            self.resolvedAlbum = album
+        }
```
[the remark this answers is not in the document]
Comment (sozercan) [reply Sources/Kaset/Views/PlayerBar.swift:911]: Addressed in c23e985 by gating eager metadata resolution on available navigation actions and resolvable artist metadata, so disabled player bars do not issue background search requests.

**Review by sozercan — APPROVED** [review]
